### ISSUE TITLE
Installation issues on Mint 17.3

### ISSUE THREAD

**Issue by qwertystop** [jcjohnson/neural-style#156]
Hi,
I'm trying to install this on a fresh install of Linux Mint 17.3. Dual-booting, not a VM, for the express purpose of running this. I tried following the instructions in install.md - I know it says they're for Ubuntu, but Mint is Ubuntu-based, yes? Unfortunately, either it's not close enough, or something else went wrong.

The clearest sign of a problem is after the `source ~/.bashrc`, which gives me a no-such-file. The th command also does not work. Prior to that, there's so much text that some of it's been pushed past the buffer before it's finished, so if there's a problem indicated somewhere in there, I can't be sure that any given one I've found is the first one.

Is there a generally-known problem with and solution for Torch on Linux Mint, or do I need to dig through the logs for something more specific?

EDIT: Trying by the install instructions on torch.ch/docs, the first two lines seem to work fine, but ./install.sh has a wide variety of warnings, "not found"s, and build errors. The last thing on the list:
```

Missing dependencies for itorch:
lbase64 
luacrypto 
uuid 
lzmq >= 0.4.2

Using https://raw.githubusercontent.com/rocks-moonscript-org/moonrocks-mirror/master/lbase64-20120820-1.src.rock... switching to 'build' mode
Do not use 'module' as a build type. Use 'builtin' instead.
gcc -O2 -fPIC -I/home/qwertystop/torch/install/include -c lbase64.c -o lbase64.o
gcc -shared -o base64.so -L/home/qwertystop/torch/install/lib lbase64.o
Updating manifest for /home/qwertystop/torch/install/lib/luarocks/rocks
lbase64 20120820-1 is now built and installed in /home/qwertystop/torch/install/ (license: Public domain)

Using https://raw.githubusercontent.com/rocks-moonscript-org/moonrocks-mirror/master/luacrypto-0.3.2-1.src.rock... switching to 'build' mode

Error: Failed installing dependency: https://raw.githubusercontent.com/rocks-moonscript-org/moonrocks-mirror/master/luacrypto-0.3.2-1.src.rock - Could not find header file for OPENSSL
  No file openssl/evp.h in /usr/local/include
  No file openssl/evp.h in /usr/include
You may have to install OPENSSL in your system and/or pass OPENSSL_DIR or OPENSSL_INCDIR to the luarocks command.
Example: luarocks install luacrypto OPENSSL_DIR=/usr/local


Not updating your shell profile.
You might want to
add the following lines to your shell profile:

. /home/qwertystop/torch/install/bin/torch-activate
```

I'm not sure what to make of that - the dependency installer seemed to work fine, and according to the Software Manager I've got openssl installed.

**Comment by htoyryla** [comment]
qwertystop <notifications@github.com> kirjoitti 5.3.2016 kello 6.16:

> Hi,
> I'm trying to install this on a fresh install of Linux Mint 17.3. Dual-booting, not a VM, for the express purpose of running this. I tried following the instructions in install.md - I know it says they're for Ubuntu, but Mint is Ubuntu-based, yes? Unfortunately, either it's not close enough, or something else went wrong.
> 
> The clearest sign of a problem is after the source ~/.bashrc, which gives me a no-such-file.
> 
This simply means that you don’t have .bashrc in your home directory. Could be that Mint doesn’t use one by default? You can create an empty one by ”touch ~/.bashrc” so that the Torch install scripts can write to it.
> The th command also does not work.
> 
Probably because of the above. Looked at my .bashrc and Torch install has written:

. /home/hannu/torch/install/bin/torch-activate

to be run every time you boot up. The ”source” command is simply used to make it run after the install without reboot. 
> Prior to that, there's so much text that some of it's been pushed past the buffer before it's finished, so if there's a problem indicated somewhere in there, I can't be sure that any given one I've found is the first one.
> 
If you have done a fresh install because of neural-style, why not Ubuntu?


Hannu

**Comment by qwertystop** [comment]
Okay, added the .bashrc. Also attempted to run torch-activate manually - no good. Currently redoing the torch installation from step 1 - going slow due to my current connection, though.

I went with Mint instead of Ubuntu because, though I installed it specifically for neural-style, I'm not going to pretend it's unlikely I'll use it for more than that with time, and having tried both, I prefer Mint. I also thought that the specific distro wouldn't matter much - that the most significant difference between two distros for this purpose was something that would be handled by the dependency-installer anyway.
